docs/bmo-part-of-capm3.md

```diff
@@ -0,0 +1,182 @@
+# Make Bare Metal Operator as part of Cluster API Provider Metal3
+
+The end goal behind making Bare Metal Operator(BMO) as part of Cluster
+API Provider Metal3(CAPM3) is to enable a successful pivoting, which requires
+making BMO to be a part of Metal3 in
+[clusterctl](https://cluster-api.sigs.k8s.io/clusterctl/commands/move.html)
```

Comment:
I would rather elaborate the motivation here a bit more . May be something in the line like we want to use clusterctl to also deploy bmo (as part of capm3) so that it can take care of moving the bmh.

Reply:
Updated